zdm/MCMC_wrap.py

```diff
@@ -0,0 +1,114 @@
+"""
+File: MCMC_wrap.py
+Author: Jordan Hoffmann
+Date: 28/09/23
+Purpose: 
+    Wrapper file to run MCMC analysis for zdm code. Handles command line
+    parameters and loading of surveys, grids and parameters. Actual MCMC
+    analysis functions are in MCMC.py.
+
+    scripts/run_mcmc.slurm contains an example sbatch script.
+"""
+
+import argparse
+import os
+
+from zdm import survey
+from zdm import cosmology as cos
+from zdm.craco import loading
+from zdm.MCMC import *
+
+import pickle
+import json
+
+parser = argparse.ArgumentParser()
+parser.add_argument(dest='files', nargs='+', help="Survey file names")
+parser.add_argument('-i','--initialise', default=None, type=str, help="Prefix used to initialise survey")
+parser.add_argument('-p','--pfile', default=None , type=str, help="File defining parameter ranges")
+parser.add_argument('-o','--opfile', default=None, type=str, help="Output file for the data")
+parser.add_argument('-w', '--walkers', default=20, type=int, help="Number of MCMC walkers")
+parser.add_argument('-s', '--steps', default=100, type=int, help="Number of MCMC steps")
+parser.add_argument('-n', '--ncpus', default=1, type=int, help="Number of CPU cores (used to determine number of parallel processes)")
+args = parser.parse_args()
+
+# Check correct flags are specified
+if args.pfile is None or args.opfile is None:
+    if not (args.pfile is None and args.opfile is None):
+        print("All flags (except -i optional) are required unless this is only for initialisation in which case only -i should be specified.")
+        exit()
+
+#==============================================================================
+
+def main(args):
+    names=args.files
+    prefix=args.initialise
+
+
+    ############## Initialise cosmology ##############
+    # Location for maximisation output
+    outdir='mcmc/'
+
+    cos.init_dist_measures()
+    state = loading.set_state()
+    
+    # get the grid of p(DM|z)
+    zDMgrid, zvals,dmvals=get_zdm_grid(state,new=True,plot=False,method='analytic',save=True,datdir='MCMCData')
+    
+    ############## Initialise surveys ##############
+
+    if not os.path.exists('Pickle/'+prefix+'surveys.pkl'):
+        # Initialise surveys
+        surveys = []
+        for name in names:
+            filename = 'data/Surveys/' + name
+            s=survey.Survey(state, name, filename, dmvals, edir='data/Efficiencies/')
+
+            surveys.append(s)
+    
+        # Initialise grids
+        grids=initialise_grids(surveys,zDMgrid, zvals,dmvals,state,wdist=True)
+
+        # Save surveys / grids in pickle format
```

Comment:
In my experience, saving surveys via Pickle is not worth it - they take up a LOT of space, and loading them might only save a minute at most. I can imagine this accidentally producing 5GB of data on somebody's computer. Could you wrap this behaviour with a bool that says "if [pickle_it] then ...", and have this bool default to false?